File: crud-item-list/src/test/java/com/gurunars/crud_item_list/CheckerSelectAllTest.java
package com.gurunars.crud_item_list;

import org.junit.Test;

import java.util.Arrays;
import java.util.HashSet;
import java.util.List;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class CheckerSelectAllTest {

    private final ActionSelectAll<String> checkerSelectAll = new ActionSelectAll<>();
    private final List<String> all = Arrays.asList("one", "two");

    @Test
    public void noSelection_leadsToTrue() throws Exception {
        assertTrue(checkerSelectAll.canPerform(all, new HashSet<String>()));
    }

    @Test
    public void selectionOfAll_leadsToFalse() throws Exception {
        assertFalse(checkerSelectAll.canPerform(all, new HashSet<>(all)));
    }

}

File: crud-item-list/src/test/kotlin/com/gurunars/crud_item_list/CheckerSelectAllTest.kt
package com.gurunars.crud_item_list

import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class CheckerSelectAllTest {

    private val checkerSelectAll = ActionSelectAll<StringItem>()
    private val all = listOf("one", "two")

    @Test
    @Throws(Exception::class)
    fun noSelection_leadsToTrue() {
        checkerSelectAll.canPerform(all, HashSet<String>(), { assertTrue(it) })
    }

    @Test
    @Throws(Exception::class)
    fun selectionOfAll_leadsToFalse() {
        checkerSelectAll.canPerform(all, HashSet(all), { assertFalse(it) })
    }
}
